EndlessClient/Dialogs/BarberDialog.cs

```diff
@@ -0,0 +1,217 @@
+            _changeBuyHairStyleOrColor.HighlightWidthOverride = InitialHighlightWidth;
+            _changeHairItem.OffsetX = AdjustedHighlightXOffset;
+            _changeHairColor.OffsetX = AdjustedHighlightXOffset;
+            _changeBuyHairStyleOrColor.OffsetX = AdjustedHighlightXOffset;
+        }
+
+        private ListDialogItem CreateListDialogItem(DialogIcon icon, int offsetY, EventHandler<MonoGame.Extended.Input.InputListeners.MouseEventArgs> clickHandler)
+        {
+            var item = new ListDialogItem(this, ListDialogItem.ListItemStyle.Large, 0)
+            {
+                OffsetY = offsetY,
+                IconGraphic = _dialogIconService.IconSheet,
+                IconGraphicSource = _dialogIconService.GetDialogIconSource(icon),
+                ShowIconBackGround = false,
+            };
+            item.LeftClick += clickHandler;
+            return item;
+        }
+
+        private XNAButton CreateButton(IEODialogButtonService dialogButtonService, Vector2 position, SmallButton buttonType)
+        {
+            var button = new XNAButton(dialogButtonService.SmallButtonSheet, position,
+                                       dialogButtonService.GetSmallDialogButtonOutSource(buttonType),
+                                       dialogButtonService.GetSmallDialogButtonOverSource(buttonType));
+            button.Initialize();
+            button.SetParentControl(this);
+            return button;
+        }
+
+        private void UpdateListItems(CharacterRenderProperties currentProperties)
+        {
+            _changeHairItem.PrimaryText = _localizedStringFinder.GetString(EOResourceID.DIALOG_BARBER_CHANGE_MODAL);
+            _changeHairItem.SubText = GetCurrentHairStyleText(currentProperties.HairStyle);
+
+            _changeHairColor.PrimaryText = _localizedStringFinder.GetString(EOResourceID.DIALOG_BARBER_CHANGE_HAIR_COLOR);
+            _changeHairColor.SubText = GetCurrentHairColorText(currentProperties.HairColor);
+
+            _changeBuyHairStyleOrColor.PrimaryText = _localizedStringFinder.GetString(EOResourceID.DIALOG_BARBER_BUY_HAIRSTYLE);
+            _changeBuyHairStyleOrColor.SubText = GetCostText();
+        }
+
+        private string GetCurrentHairStyleText(int hairStyle) => $"{_localizedStringFinder.GetString(EOResourceID.DIALOG_WORD_CURRENT)}: {hairStyle}";
+        private string GetCurrentHairColorText(int hairColor) => $"{_localizedStringFinder.GetString(EOResourceID.DIALOG_WORD_CURRENT)}: {_hairColorNames[hairColor % _hairColorNames.Length]}";
+        private string GetCostText() => $"{_localizedStringFinder.GetString(EOResourceID.DIALOG_WORD_CURRENT)}: {CalculateCost()} gold";
+
+        private int CalculateCost()
+        {
+            var level = (int)_characterRepository.MainCharacter.Stats[CharacterStat.Level];
+            return 200 + Math.Max(level - 1, 0) * 200;
+        }
+
+        private void Hair(object sender, MonoGame.Extended.Input.InputListeners.MouseEventArgs e)
```

Comment:
Rename the event handlers something like `ChangeHairStyle_Click`, `ChangeHairColor_Click`, `BuyHair_Click` so it's clearer from the names that they're vent handlers.